### PR TITLE
Gitlab CI

### PR DESCRIPTION
**Pull request by metadave**
- before merge, remove the `dp_gitlab_ci` tag from the stage `only` section
- the deployment Dockerfile is in ./bin, using a Dockerfile with a different suffix results in docker unhappiness
- [example build](https://gitlab.com/mozmeao/lumbergh/pipelines/53214483)

There is also a scheduled cronjob that needs to be updated:
https://gitlab.com/mozmeao/lumbergh/pipeline_schedules

### CODE REVIEW

File: bin/ci-mirror.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash -e
+
+# Build an image that renders and copies static content to an S3 bucket
+# Ping DMS when complete
+
+export IMAGE=mozmeao/lumbergh-deploy:`git rev-parse --short HEAD`
+docker build -f ./bin/Dockerfile . -t ${IMAGE}
+docker run \
+      -e AWS_ACCESS_KEY_ID="${AWS_ACCESS_KEY_ID}" \
```
Comment (pmac) [inline bin/ci-mirror.sh]: NIT: If the env var is in the environment, you can just `-e AWS_ACCESS_KEY_ID` and [it'll use the value of that var in the env](https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file). Might make things easier for next time.

File: bin/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+# First copy requirements.txt and peep so we can take advantage of
+# docker caching.
+COPY requirements.txt requirements.txt
+RUN pip install --require-hashes --no-cache-dir -r requirements.txt
+RUN pip install awscli
+
+COPY --chown=webdev . /app
+USER webdev
+CMD ./bin/mirror.sh && aws s3 sync ./_site ${BUCKET_PATH} --acl public-read --delete && curl ${DMS}
```
Comment (pmac) [inline bin/Dockerfile]: I'd add all of that to a bash script, but it's not required.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -1,11 +1,30 @@
...
+# DMS DeadMansSnitch URL
+# PROD_BUCKET_PATH (w/ s3:// prefix)
+# STAGE_BUCKET_PATH (ex: s3://mozilla-careers-stage)
+
+deploy stage:
+  stage: deploy
+  tags: 
+  - meao
+  variables:
+    BUCKET_PATH: ${STAGE_BUCKET_PATH}
```
Comment (pmac) [inline .gitlab-ci.yml]: do the bucket paths have secrets? If not you can add them to this file in a `variables:` directive and they'll be available to all jobs.

File: bin/ci-mirror.sh (hunk 1 above)
Comment (metadave) [reply bin/ci-mirror.sh]: nice, I'll use that syntax.

File: .gitlab-ci.yml (hunk 3 above)
Comment (metadave) [reply .gitlab-ci.yml]: no, there's nothing secret so I'm switching to `variables`.

**Pull request merged**